libraries/botbuilder-lg/src/mslgTool.ts

```diff
@@ -31,13 +30,14 @@ export class MSLGTool {
     }
 
     public validateFile(lgFileContent: string, id?: string): string[] {
-        const diagnostic: Diagnostic[] = new StaticChecker().checkText(lgFileContent, id, ImportResolver.fileResolver);
+        const lgFile = LGParser.parseText(lgFileContent, id);
+        const diagnostic: Diagnostic[] = new StaticChecker(lgFile).check();
```

Comment:
we can simply use the lgFile.diagnostics to check the errors, there is no need to use StaticChecker